flow/layers/aiks_layer.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/aiks_layer.h"
+
+#include <utility>
+
+#include "flutter/display_list/dl_builder.h"
+#include "flutter/flow/layer_snapshot_store.h"
+#include "flutter/flow/layers/cacheable_layer.h"
+#include "flutter/flow/layers/offscreen_surface.h"
+#include "flutter/flow/raster_cache.h"
+#include "flutter/flow/raster_cache_util.h"
+
+namespace flutter {
+
+AiksLayer::AiksLayer(const SkPoint& offset,
+                     const std::shared_ptr<const impeller::Picture>& picture,
+                     bool is_complex,
+                     bool will_change)
+    : offset_(offset), picture_(picture) {
+#if IMPELLER_SUPPORTS_RENDERING
+  if (picture_) {
+    auto bounds = picture_->pass->GetElementsCoverage(std::nullopt)
+                      .value_or(impeller::Rect());
+    bounds_ = SkRect::MakeXYWH(bounds.origin.x + offset_.x(),
+                               bounds.origin.y + offset.y(), bounds.size.width,
+                               bounds.size.height);
+  }
+#endif
+}
+
+void AiksLayer::Diff(DiffContext* context, const Layer* old_layer) {
+  DiffContext::AutoSubtreeRestore subtree(context);
+  context->PushTransform(SkMatrix::Translate(offset_.x(), offset_.y()));
+  context->AddLayerBounds(bounds_);
+  context->SetLayerPaintRegion(this, context->CurrentSubtreeRegion());
+}
+
+void AiksLayer::Preroll(PrerollContext* frame) {
+  // TODO(dnfield): Handle group opacity checks.
+  set_paint_bounds(bounds_);
```

Comment:
Link to an issue?

Reply:
I'm going to delete this comment.

In `DisplayListLayer::Preroll`, there's opacity related checking that happens to make decisions about raster caching. Raster caching isn't a thing for Impeller, and Impeller internally handles its own opacity peepholing.

Reply:
(adding a comment about that and a dcheck that there's no raster cache)